homeassistant/components/wittiot/sensor.py

```diff
@@ -0,0 +1,1233 @@
+TYPE_LEAKCH2 = "leak_ch2"
+TYPE_LEAKCH3 = "leak_ch3"
+TYPE_LEAKCH4 = "leak_ch4"
+TYPE_TEMPCH1 = "temp_ch1"
+TYPE_TEMPCH2 = "temp_ch2"
+TYPE_TEMPCH3 = "temp_ch3"
+TYPE_TEMPCH4 = "temp_ch4"
+TYPE_TEMPCH5 = "temp_ch5"
+TYPE_TEMPCH6 = "temp_ch6"
+TYPE_TEMPCH7 = "temp_ch7"
+TYPE_TEMPCH8 = "temp_ch8"
+TYPE_HUMICH1 = "humidity_ch1"
+TYPE_HUMICH2 = "humidity_ch2"
+TYPE_HUMICH3 = "humidity_ch3"
+TYPE_HUMICH4 = "humidity_ch4"
+TYPE_HUMICH5 = "humidity_ch5"
+TYPE_HUMICH6 = "humidity_ch6"
+TYPE_HUMICH7 = "humidity_ch7"
+TYPE_HUMICH8 = "humidity_ch8"
+TYPE_SOILCH1 = "Soilmoisture_ch1"
+TYPE_SOILCH2 = "Soilmoisture_ch2"
+TYPE_SOILCH3 = "Soilmoisture_ch3"
+TYPE_SOILCH4 = "Soilmoisture_ch4"
+TYPE_SOILCH5 = "Soilmoisture_ch5"
+TYPE_SOILCH6 = "Soilmoisture_ch6"
+TYPE_SOILCH7 = "Soilmoisture_ch7"
+TYPE_SOILCH8 = "Soilmoisture_ch8"
+TYPE_ONLYTEMPCH1 = "tf_ch1"
+TYPE_ONLYTEMPCH2 = "tf_ch2"
+TYPE_ONLYTEMPCH3 = "tf_ch3"
+TYPE_ONLYTEMPCH4 = "tf_ch4"
+TYPE_ONLYTEMPCH5 = "tf_ch5"
+TYPE_ONLYTEMPCH6 = "tf_ch6"
+TYPE_ONLYTEMPCH7 = "tf_ch7"
+TYPE_ONLYTEMPCH8 = "tf_ch8"
+TYPE_LEAFCH1 = "leaf_ch1"
+TYPE_LEAFCH2 = "leaf_ch2"
+TYPE_LEAFCH3 = "leaf_ch3"
+TYPE_LEAFCH4 = "leaf_ch4"
+TYPE_LEAFCH5 = "leaf_ch5"
+TYPE_LEAFCH6 = "leaf_ch6"
+TYPE_LEAFCH7 = "leaf_ch7"
+TYPE_LEAFCH8 = "leaf_ch8"
+
+TYPE_PM25CH1_BATT = "pm25_ch1_batt"
+TYPE_PM25CH2_BATT = "pm25_ch2_batt"
+TYPE_PM25CH3_BATT = "pm25_ch3_batt"
+TYPE_PM25CH4_BATT = "pm25_ch4_batt"
+TYPE_LEAKCH1_BATT = "leak_ch1_batt"
+TYPE_LEAKCH2_BATT = "leak_ch2_batt"
+TYPE_LEAKCH3_BATT = "leak_ch3_batt"
+TYPE_LEAKCH4_BATT = "leak_ch4_batt"
+TYPE_TEMPCH1_BATT = "temph_ch1_batt"
+TYPE_TEMPCH2_BATT = "temph_ch2_batt"
+TYPE_TEMPCH3_BATT = "temph_ch3_batt"
+TYPE_TEMPCH4_BATT = "temph_ch4_batt"
+TYPE_TEMPCH5_BATT = "temph_ch5_batt"
+TYPE_TEMPCH6_BATT = "temph_ch6_batt"
+TYPE_TEMPCH7_BATT = "temph_ch7_batt"
+TYPE_TEMPCH8_BATT = "temph_ch8_batt"
+TYPE_SOILCH1_BATT = "Soilmoisture_ch1_batt"
+TYPE_SOILCH2_BATT = "Soilmoisture_ch2_batt"
+TYPE_SOILCH3_BATT = "Soilmoisture_ch3_batt"
+TYPE_SOILCH4_BATT = "Soilmoisture_ch4_batt"
+TYPE_SOILCH5_BATT = "Soilmoisture_ch5_batt"
+TYPE_SOILCH6_BATT = "Soilmoisture_ch6_batt"
+TYPE_SOILCH7_BATT = "Soilmoisture_ch7_batt"
+TYPE_SOILCH8_BATT = "Soilmoisture_ch8_batt"
+TYPE_ONLYTEMPCH1_BATT = "tf_ch1_batt"
+TYPE_ONLYTEMPCH2_BATT = "tf_ch2_batt"
+TYPE_ONLYTEMPCH3_BATT = "tf_ch3_batt"
+TYPE_ONLYTEMPCH4_BATT = "tf_ch4_batt"
+TYPE_ONLYTEMPCH5_BATT = "tf_ch5_batt"
+TYPE_ONLYTEMPCH6_BATT = "tf_ch6_batt"
+TYPE_ONLYTEMPCH7_BATT = "tf_ch7_batt"
+TYPE_ONLYTEMPCH8_BATT = "tf_ch8_batt"
+TYPE_LEAFCH1_BATT = "leaf_ch1_batt"
+TYPE_LEAFCH2_BATT = "leaf_ch2_batt"
+TYPE_LEAFCH3_BATT = "leaf_ch3_batt"
+TYPE_LEAFCH4_BATT = "leaf_ch4_batt"
+TYPE_LEAFCH5_BATT = "leaf_ch5_batt"
+TYPE_LEAFCH6_BATT = "leaf_ch6_batt"
+TYPE_LEAFCH7_BATT = "leaf_ch7_batt"
+TYPE_LEAFCH8_BATT = "leaf_ch8_batt"
+
+
+@dataclass(frozen=True)
+class WittiotSensorEntityDescription(SensorEntityDescription):
+    """Class describing WittIOT sensor entities."""
```

Comment:
can't we just add a custom key `sensor_type` here and avoid all the special arrays and `elif:`s? (You will need to add `kw_only=True` to the `@dataclass`)